DB/Item.lua

```diff
@@ -165,4 +165,83 @@ function Item:HasZone(entry)
 	return false
 end
 
+function Item:IsCollectionItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.COLLECTION
+end
+
+function Item:IsUsingCollectionProperties(entry)
+	return entry.collectedItemId ~= nil
+end
+
+function Item:IsArchaeologyItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.ARCH
+end
+
+function Item:IsUsingArchaeologyProperties(entry)
+	return entry.raceId ~= nil
+end
+
+function Item:HasUseMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.USE
+end
+
+function Item:HasUseMethodProperties(entry)
+	return entry.items ~= nil
+end
+
+function Item:IsCovenantSpecificItem(entry)
+	return entry.requiresCovenant ~= nil
+end
+
+function Item:IsUsingCovenantProperties(entry)
+	return entry.requiredCovenantID ~= nil
+end
+
+function Item:IsZoneItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.ZONE
+end
+
+function Item:IsUsingZoneProperties(entry)
+	return entry.zones ~= nil
+end
+
+function Item:IsUsingFishingProperties(entry)
+	return entry.requiresPool ~= nil
+end
+
+function Item:IsFishingItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.FISHING
+end
+
+function Item:HasNPCMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.NPC
+end
+
+function Item:HasBossMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.BOSS
+end
+
+function Item:IsUsingNPCProperties(entry)
+	return entry.npcs ~= nil
+end
+
+function Item:HasSpecialMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.SPECIAL
+end
+
+function Item:IsUsingCoordsProperties(entry)
```

Comment:
``CoordsProperties`` is technically accurate, but a better name might be ``HasWaypoints`` or ``HasWaypointsData``?